runtime/tests/decimal_arith.c

```diff
@@ -135,10 +135,48 @@ void testDiv() {
     genAndValidateDecDiv("9.999999999999999999999999999999999E-6001", "0.5E143", "2.000000000000000000000000000000000E-6143", 0);
 }
 
+void genAndValidateDecNeg(const char *dec, const char *decMinus) {
+    TaggedPtr tp = _bal_decimal_neg(_bal_decimal_const(dec));
+    validateDec(tp, decMinus);
+}
+
+void testNeg() {
+    genAndValidateDecNeg("0", "0");
+    genAndValidateDecNeg("0.0", "0.0");
+    genAndValidateDecNeg("1", "-1");
+    genAndValidateDecNeg("-1", "1");
+    genAndValidateDecNeg("1E-6143", "-1E-6143");
+    genAndValidateDecNeg("99999999999999999999999999999999999", "-1.000000000000000000000000000000000E+35");
+    genAndValidateDecNeg("9.999999999999999999999999999999998E6144", "-9.999999999999999999999999999999998E+6144");
+}
+
+void genAndValidateDecRem(const char *decStr1, const char *decStr2, const char *rem, PanicCode code) {
+    TaggedPtrPanicCode tp = _bal_decimal_rem(_bal_decimal_const(decStr1), _bal_decimal_const(decStr2));
+    validate(tp, rem, code);
+}
+
+void testRem() {
+    genAndValidateDecRem("1", "1", "0", 0);
```

Comment:
Need tests whether the second operand is negative.

Reply:
Make it obvious that these are giving the same results as integer division by using eg `(-10%3).toString()`

Reply:
Added more tests

Reply:
I want to be absolutely sure that we are not getting the IEEE Remainder operation. Which are the tests that show that? Are the decNumber docs clear on this?  See this for some concrete cases 

https://docs.microsoft.com/en-us/dotnet/api/system.math.ieeeremainder?view=net-6.0